src/server/pkg/migrations/migrations.go

```diff
@@ -68,28 +68,44 @@ func InitialState() State {
 	return State{
 		name: "init",
 		change: func(ctx context.Context, env Env) error {
-			_, err := env.Tx.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS migrations (
-				id BIGINT PRIMARY KEY,
-				NAME VARCHAR(250) NOT NULL,
-				start_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
-				end_time TIMESTAMP
-			);
-			INSERT INTO migrations (id, name, start_time, end_time) VALUES (0, 'init', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP) ON CONFLICT DO NOTHING;
-			`)
-			return err
+			return nil
 		},
 	}
 }
 
 // ApplyMigrations does the necessary work to actualize state.
 // It will manipulate the objects available in baseEnv, and use the migrations table in db.
 func ApplyMigrations(ctx context.Context, db *sqlx.DB, baseEnv Env, state State) error {
+	tx, err := db.BeginTxx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+
+	env := baseEnv
+	env.Tx = tx
+
+	if _, err := tx.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS migrations (
```

Comment:
I'm sort of indifferent as to whether it's easier to special case the creation of the table here or in the initial state change.  We need to special case isFinished of state 0 to check for the existence of the table, and special case the execution of state 0 to create the table, without the usual locking.

So this shouldn't be executed during every call to ApplyMigration, it should get skipped by isFinished in the common case.

I want to avoid special-casing state 0 to callers of this API.  It should represent whether the migrations table has been created. If it has been actualized, then the table exists, if not then it hasn't.  We shouldn't have behavior where InitialState is a thing you have to call to use the API, but the state you get back is actually totally meaningless because the creation of the table doesn't correspond to state 0's actualization.